locked-nft/contracts/NFTLocker.cdc

```diff
@@ -89,27 +93,177 @@ access(all) contract NFTLocker {
         return false
     }
 
-    /// The path to the NFTLocker Admin resource belonging to the Account
-    /// which the contract is deployed on
-    access(all) view fun GetAdminStoragePath(): StoragePath {
+    /// The path to the Admin resource belonging to the account where this contract is deployed
+    ///
+    access(all) view fun getAdminStoragePath(): StoragePath {
         return /storage/NFTLockerAdmin
     }
 
+    /// The path to the ReceiverCollector resource belonging to the account where this contract is deployed
+    ///
+    access(all) view fun getReceiverCollectorStoragePath(): StoragePath {
+        return /storage/NFTLockerAdminReceiverCollector
+    }
+
+    /// Return an unauthorized reference to the admin's ReceiverCollector resource if it exists
+    ///
+    access(all) view fun borrowAdminReceiverCollectorPublic(): &ReceiverCollector? {
+        return self.account.storage.borrow<&ReceiverCollector>(from: NFTLocker.getReceiverCollectorStoragePath())
+    }
+
+    /// Interface for depositing NFTs to authorized receivers
+    ///
+    access(all) struct interface IAuthorizedDepositHandler {
+        access(all) fun deposit(nft: @{NonFungibleToken.NFT}, ownerAddress: Address, passThruParams: {String: AnyStruct})
+    }
+
+    /// Struct that defines a Receiver
+    ///
+    /// Receivers are entities that can receive locked NFTs and deposit them using a specific deposit method
+    ///
+    access(all) struct Receiver {
+        /// Handler for depositing NFTs for the receiver
+        ///
+        access(all) var authorizedDepositHandler: {IAuthorizedDepositHandler}
+
+        /// The eligible NFT types for the receiver
+        ///
+        access(all) let eligibleNFTTypes: {Type: Bool}
+
+        /// Extension map for additional data
+        ///
+        access(all) let metadata: {String: AnyStruct}
+
+        /// Initialize Receiver struct
+        ///
+        view init(
+            authorizedDepositHandler: {IAuthorizedDepositHandler},
+            eligibleNFTTypes: {Type: Bool}
+        ) {
+            self.authorizedDepositHandler = authorizedDepositHandler
+            self.eligibleNFTTypes = eligibleNFTTypes
+            self.metadata = {}
+        }
+    }
+
+    /// ReceiverCollector resource
+    ///
+    /// Note: This resource is used to store receivers and corresponding authorized deposit handlers; currently,
+    /// only the admin account can add or remove receivers - in the future, a ReceiverProvider resource could
+    /// be added to provide this capability to separate authorized accounts.
+    ///
+    access(all) resource ReceiverCollector  {
```

Comment:
New admin resource to store authorized deposit handlers - this allows separating concerns with the existing `Admin` resource, and it's also not possible to add new fields into the existing `Admin` resource (not a valid change for contract update).

 The resource is meant to be stored at a specific path in the admin account storage to allow certain methods to be accessible publicly via `borrowAdminReceiverCollectorPublic`.